src/commands/plug.rs

```diff
@@ -0,0 +1,132 @@
+use std::{io::Write as _, path::PathBuf};
+
+use anyhow::{bail, Context as _, Result};
+use clap::Args;
+use wac_graph::{CompositionGraph, EncodeOptions, NodeId, PackageId};
+use wac_types::{Package, SubtypeChecker};
+
+/// Plugs the exports of any number of 'plug' components into the imports of a 'socket' component.
+#[derive(Args)]
+#[clap(disable_version_flag = true)]
+pub struct PlugCommand {
+    /// The path to the plug component.
+    ///
+    /// More than one plug can be supplied.
+    #[clap(long = "plug", value_name = "PLUG_PATH", required = true)]
+    pub plugs: Vec<PathBuf>,
+
+    /// The path to the socket component
+    #[clap(value_name = "SOCKET_PATH", required = true)]
+    pub socket: PathBuf,
+
+    /// The path to write the output to.
+    ///
+    /// If not specified, the output will be written to stdout.
+    #[clap(long, short = 'o')]
+    pub output: Option<PathBuf>,
+}
+
+impl PlugCommand {
+    /// Executes the command.
+    pub async fn exec(&self) -> Result<()> {
+        log::debug!("executing plug command");
+        let mut graph = CompositionGraph::new();
+
+        let socket = std::fs::read(&self.socket).with_context(|| {
+            format!(
+                "failed to read socket component `{path}`",
+                path = self.socket.display()
+            )
+        })?;
+
+        let socket = Package::from_bytes("socket", None, socket, graph.types_mut())?;
+        let socket = graph.register_package(socket)?;
+        let socket_instantiation = graph.instantiate(socket);
+
+        // Plug each plug into the socket.
+        for (i, plug) in self.plugs.iter().enumerate() {
+            let plug = std::fs::read(&plug).with_context(|| {
+                format!(
+                    "failed to read plug component `{path}`",
+                    path = plug.display()
+                )
+            })?;
+            plug_into_socket(
+                &format!("plug{i}"),
+                plug,
+                socket,
+                socket_instantiation,
+                &mut graph,
+            )?;
+        }
+
+        // Check we've actually done any plugging.
+        if graph
+            .get_instantiation_arguments(socket_instantiation)
+            .next()
+            .is_none()
+        {
+            bail!("no plugs were used to plug into the socket component")
```

Comment:
```suggestion
            bail!("the socket component had no matching imports for the plugs that were provided")
```

Reply:
👍, but I wonder if this should be a warning rather than an error.

Reply:
I think we can keep it as an error for now unless someone really wants wac to wrap the socket component with an instantiation and re-export of everything it exports (an entirely wasteful operation).

Reply:
Sounds good. I'm just thinking it might happen in scenarios with variable content where sometimes a composition happens and sometimes not. If we keep it an error we might force people to check before hand that a composition will happen instead of just silently doing nothing.